CMakeLists.txt

```diff
@@ -13,13 +13,11 @@ find_program(YAML_CPP_CLANG_FORMAT_EXE NAMES clang-format)
 option(YAML_CPP_BUILD_CONTRIB "Enable yaml-cpp contrib in library" ON)
 option(YAML_CPP_BUILD_TOOLS "Enable parse tools" ON)
 option(YAML_BUILD_SHARED_LIBS "Build yaml-cpp shared library" ${BUILD_SHARED_LIBS})
+option(YAML_CPP_INSTALL "Enable generation of yaml-cpp install targets" ON)
```

Comment:
This will always install yaml-cpp, also when included as a subproject, which is often undesired.
I would change the default to only install when yaml-cpp is the main project.

So something as follows:

Add this to the top of the cmake script:
```cmake
set(YAML_CPP_MAIN_PROJECT ON)
if(DEFINED PROJECT_SOURCE_DIR
    set(YAML_CPP_MAIN_PROJECT OFF)
endif()

# 3.5 is actually available almost everywhere, but this a good minimum
cmake_minimum_required(VERSION 3.4)
project(YAML_CPP VERSION 0.6.3 LANGUAGES CXX)
```

And then change this line to:
```suggestion
option(YAML_CPP_INSTALL "Enable generation of yaml-cpp install targets" ${YAML_CPP_MAIN_PROJECT})
```

This combines the best of both world:
- yaml-cpp is not installed (by default) when included as a subproject
- if it is really wanted, yaml-cpp can be installed by enabling the `YAML_CPP_INSTALL` cache definition.